src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -12,7 +12,7 @@
 )
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="function")
```

Comment:
this will severely slow down tests. what's the justification? the update tests are written in a way where this fixture can be created once before tests run and the same fixture is used in all tests

Reply:
Cortez and I were running into an error for contract_update_create_only_property(), where it's automatically failing on the [create call](https://github.com/aws-cloudformation/cloudformation-cli/blob/0d71a6520ba364c94301f6b8659c203ed4b4d5a3/src/rpdk/core/contract/suite/handler_update.py#L63) because the resource from the previous test function is never cleaned up since the fixture is using a modular scope

Reply:
If you are worried about slowing  the test down, we could rip these 2 test: contract_update_create_only_property and  contract_update_non_existent_resource into another class which does not have a fixture.

Reply:
As far as I understand now, the fixture is being created twice instead of once, as only 2 tests require it. But there is a risk of the number of tests increasing in this class causing the tests to slow down

Reply:
I think that the update tests are a bit special in that they are testing multiple lifecycle event happening on a resource (because we inherently have to do creates first). There is not really a way to test _only_ the update functionality but rather the interplay of create-update.